homeassistant/components/qvrpro/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""Support for QVR Pro NVR software by QNAP."""
+
+import logging
+
+from pyqvrpro import Client
+from pyqvrpro.client import AuthenticationError, InsufficientPermissionsError
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+
+from .const import CONF_EXCLUDE_CHANNELS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_HOST): cv.string,
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Optional(CONF_EXCLUDE_CHANNELS, default=[]): vol.All(
+                    cv.ensure_list_csv, [cv.positive_int]
+                ),
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+def setup(hass, config):
+    """Set up the QVR Pro component."""
+    user = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+    host = config[DOMAIN][CONF_HOST]
+    excluded_channels = config[DOMAIN][CONF_EXCLUDE_CHANNELS]
```

Comment:
You can leave [DOMAIN] out.

Reply:
The config object passed into the setup method is not limited to qvr specific configuration items.  Removing the domain key breaks config lookup.

![image](https://user-images.githubusercontent.com/453840/74860474-4e81d300-530e-11ea-9078-c05ae0969265.png)

Reply:
Then please introduce a new variable beforehand called conf, e.g. `conf = config[DOMAIN]`.